sky/engine/core/script/dart_init.cc

```diff
@@ -351,6 +353,10 @@ static void EmbedderTimelineStopRecording() {
   callbacks->stop_tracing_callback.Run();
 }
 
+void SetServiceIsolateHook(ServiceIsolateHook hook) {
```

Comment:
I am bit nervous about this call since it is so sensitive to timing. Depending on whether the service isolate has been started already, my hook may or may not be called. Can we somehow ensure that if the hook is set too late, the caller knows that his callback may never be invoked (assert is probably fine).